### PR TITLE
rpc: RPCHelpMan: Always name dictionary keys

### PR DESCRIPTION
**Pull request by maflcko**
Fixes two issues reported in #15737:

* > I am very perplexed as to how the code I'm looking at is generating the help text I'm seeing

So add documentation

* > This is a value for which a key is missing

So always serialize the name of the dictionary key if the outer type is a dictionary

### CODE REVIEW

**Comment by MarcoFalke** [comment]
This bugfix can not be tested, because none of our RPCs accept nested dictionaries. However, the following diff might help:

```diff
diff --git a/src/wallet/rpcdump.cpp b/src/wallet/rpcdump.cpp
index 9c5dae3623..81d282ceca 100644
--- a/src/wallet/rpcdump.cpp
+++ b/src/wallet/rpcdump.cpp
@@ -1377,7 +1377,7 @@ UniValue importmulti(const JSONRPCRequest& mainRequest)
                                             {"pubKey", RPCArg::Type::STR, RPCArg::Optional::OMITTED, ""},
                                         }
                                     },
-                                    {"keys", RPCArg::Type::ARR, /* default */ "empty array", "Array of strings giving private keys to import. The corresponding public keys must occur in the output or redeemscript.",
+                                    {"keys", RPCArg::Type::OBJ, /* default */ "empty array", "Array of strings giving private keys to import. The corresponding public keys must occur in the output or redeemscript.",
                                         {
                                             {"key", RPCArg::Type::STR, RPCArg::Optional::OMITTED, ""},
                                         }
```

With the above diff, but before and after my bugfix:

```diff
diff --git a/importmulti b/importmulti
index 7a79198..d2eb7e7 100644
--- a/importmulti
+++ b/importmulti
@@ -25,7 +25,7 @@ Arguments:
            "pubKey",                                                (string)
            ...
          ],
-         {                                                          (json object, optional, default=empty array) Array of strings giving private keys to import. The corresponding public keys must occur in the output or redeemscript.
+         "keys": {                                                  (json object, optional, default=empty array) Array of strings giving private keys to import. The corresponding public keys must occur in the output or redeemscript.
            "key": "str",                                            (string)
          },
          "range": n or [n,n],                                       (numeric or array) If a ranged descriptor is used, this specifies the end or the range (in the form [begin,end]) to import

**Comment by gwillen** [comment]
Aha, thanks -- this looks exactly like the problem I was seeing with "range", which _was_ a nested dictionary at the time of my last rebase, even though it didn't exist in 0.17, and is a nested array on master.

Kudos for finding the problem despite the repro being missing. :-)

**Comment by gwillen** [comment]
utACK

**Comment by promag** [comment]
Tested ACK fa26eb5.

File: src/rpc/util.cpp [hunk 1]
```diff
@@ -213,7 +231,7 @@ struct Sections {
         case RPCArg::Type::OBJ:
         case RPCArg::Type::OBJ_USER_KEYS: {
             const auto right = outer_type == OuterType::NAMED_ARG ? "" : arg.ToDescriptionString();
-            PushSection({indent + "{", right});
+            PushSection({indent + (push_name ? "\"" + arg.m_name + "\": " : "") + "{", right});
```
Comment (promag) [inline src/rpc/util.cpp]: nit,
```
const auto left = push_name ? "\"" + arg.m_name + "\": " : "";
PushSection({indent + left, right);
```

**Pull request merged**